### PR TITLE
fix: add exception to catch session not having JWT

### PR DESCRIPTION
**Pull request by hughhhh**
### SUMMARY

Catching the following exception to return 401 instead of 500.
```
flask_jwt_extended.exceptions.NoAuthorizationError: Missing JWT in cookies or headers (Missing cookie "__s__"; Missing Authorization Header
```

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/views/base.py [hunk 1]
```diff
@@ -166,6 +167,9 @@ def api(f: Callable[..., FlaskResponse]) -> Callable[..., FlaskResponse]:
     def wraps(self: "BaseSupersetView", *args: Any, **kwargs: Any) -> FlaskResponse:
         try:
             return f(self, *args, **kwargs)
+        except NoAuthorizationError as ex: # pylint: disable=broad-except
+            logger.exception(ex)
```
Comment (suddjian) [inline superset/views/base.py:171]: We probably don't want to log this as an exception, since it's not a server error.

Comment (nytai) [reply superset/views/base.py:171]: +1, lets just log as a warning since this isn't really actionable

File: superset/views/base.py [hunk 2]
```diff
@@ -30,6 +30,7 @@
 from flask_appbuilder.models.sqla.filters import BaseFilter
 from flask_appbuilder.security.sqla.models import Role, User
 from flask_appbuilder.widgets import ListWidget
+from flask_jwt_extended.exceptions import NoAuthorizationError
```
Comment (nytai) [inline superset/views/base.py:33]: hmm, I think this package is optional, it might inherit from a generic flask auth exception class

Comment (nytai) [reply superset/views/base.py:33]: 😭 looks like it's just inheriting from `Exception`

**Review by suddjian — APPROVED** [review]
Thanks for the fix!

**Pull request merged**